src/DocLayer.actor.cpp

```diff
@@ -165,10 +165,23 @@ ACTOR Future<Void> popDisposedMessages(Reference<BufferedConnection> bc,
 	}
 }
 
+// Splitting the delay into smaller delay(). For each delay(), DelayedTask is created and added
+// to the queue. This task stays in queue until delay is expired, even if all the futures are
+// cancelled. In the connection churn it is possible we are leaving too many long delays
+// in the queues.
+ACTOR Future<Void> delayCursorExpiry() {
+	state int remaining = DOCLAYER_KNOBS->CURSOR_EXPIRY;
+	while (remaining > 0) {
+		wait(delay(1.0));
+		remaining--;
+	}
+	return Void();
+}
+
 ACTOR Future<Void> housekeeping(Reference<ExtConnection> ec) {
 	try {
 		loop {
-			wait(delay(DOCLAYER_KNOBS->CURSOR_EXPIRY));
```

Comment:
Will `delayCursorExpiry()` be different from the original `delay(DOCLAYER_KNOBS->CURSOR_EXPIRY)`?

Reply:
It has same affect when actor is not cancelled. But, when the actor is cancelled, it gets destroyed sooner than simple `delay()`.

Reply:
I failed to see how this can make the cancel faster, but I don't think it will make anything worse either.

Reply:
Probably my choice of words is not the best. It doesn't make `cancel` faster. Every `delay()` adds a `DelayedTask` in the queue. Even when the all the futures for delay() are gone, this `DelayedTask` stays there until `delay()`duration is completed. If you have too many of these actors getting cancelled with long `delay()`s then timer queue in `Net2.actor.cpp` would be full of these objects. If you have huge connection churn, with long delays this could take up too much memory.

Reply:
Ah I see. It's to reduce the memory usage. Got it.